slack_block.py

```diff
@@ -0,0 +1,85 @@
+from slacker import Slacker
+
+from nio.common.block.base import Block
+from nio.common.discovery import Discoverable, DiscoverableType
+from nio.metadata.properties import VersionProperty, StringProperty, \
+    ExpressionProperty, ObjectProperty, PropertyHolder
+
+
+class BotInformation(PropertyHolder):
+
+    name = ExpressionProperty(title='Name', default='')
+    icon = ExpressionProperty(
+        title='Icon (URL or emoji)', default=':smiley_cat:')
+
+    def get_bot_details(self, signal):
+        """ Returns a dictionary of kwargs for the bot details.
+
+        Given a signal, this method will construct a dictionary of kwargs
+        for the Slack API call with the correct parameters. Mainly, it will
+        figure out whether an icon is a URL or an emoji and use the correct
+        parameter name
+        """
+        param_dict = dict()
+
+        # Put the bot's name in the username field
+        param_dict['username'] = self.name(signal)
+
+        # If it starts and ends with a colon, it's an emoji code
+        icon = self.icon(signal).strip()
+        if icon.startswith(':') and icon.endswith(':'):
+            param_dict['icon_emoji'] = icon
+        else:
+            param_dict['icon_url'] = icon
+
+        return param_dict
+
+
+@Discoverable(DiscoverableType.block)
+class Slack(Block):
+
+    """ Send messages to a slack channel as a bot """
+
+    version = VersionProperty('0.1.0')
+    api_token = StringProperty(
+        title='Slack API Token', default='[[SLACK_API_TOKEN]]')
+    channel = ExpressionProperty(title='Slack Channel', default='{{$channel}}',
```

Comment:
I have started to prefer to put spaces in my expressions. `{{ $channel }}` 

It's just a personal preference but we should talk about this and be more consistent.